src/interactions/Commands/ChatInputCommands/devFunctions.ts

```diff
@@ -0,0 +1,444 @@
+	} else {
+		discordCommand = await interaction.client.application?.commands.fetch().then((cList) => cList.find((c) => c.name === commandNames.command));
+	}
+
+	const targetUser = interaction.options.getUser("user", true);
+	const targetMember = await interaction.guild?.members.fetch(targetUser.id).catch(() => null);
+	const target = {
+		user: {
+			username: targetUser?.username,
+			public_flags: targetUser?.flags?.bitfield,
+			id: targetUser.id,
+			discriminator: targetUser.discriminator,
+			avatar: targetUser.avatar,
+		},
+		roles: Array.isArray(targetMember?.roles) ? targetMember?.roles : targetMember?.roles.cache.map((r) => r.id),
+		premium_since: targetMember?.premiumSinceTimestamp,
+		permissions: targetMember?.permissions.bitfield,
+		pending: targetMember?.pending,
+		nick: targetMember?.nickname,
+		mute: targetMember?.voice.mute,
+		joined_at: targetMember?.joinedAt?.toISOString(),
+		is_pending: targetMember?.pending,
+		flags: 0,
+		deaf: targetMember?.voice.deaf,
+		communication_disabled_until: targetMember?.communicationDisabledUntilTimestamp,
+		avatar: targetMember?.avatar,
+	};
+
+	const resolved: {
+		members: { [key: string]: GuildMemberResolvable };
+		users: { [key: string]: UserResolvable };
+		channels: { [key: string]: ChannelResolvable };
+	} = { members: {}, users: {}, channels: {} };
+
+	const formattedOptions = await Promise.all(
+		options.map(async (o) => {
+			const option: { name: string; type: ApplicationCommandOptionType; value: string; user?: User; member?: GuildMember } = {
+				name: o.name,
+				type: o.type,
+				value: o.value,
+			};
+
+			if (o.type === ApplicationCommandOptionType.User) {
+				const user = await client.users.fetch(o.value);
+				resolved.users[o.value] = user;
+				const member = await interaction.guild?.members.fetch(o.value);
+				if (!!member) resolved.members[o.value] = member;
+
+				option.user = user;
+				option.member = member;
+			} else if (o.type === ApplicationCommandOptionType.Channel) {
+				const channel = await client.channels.fetch(o.value);
+				if (!!channel) resolved.channels[o.value] = channel;
+			}
+
+			console.log(option);
+			return option;
+		})
+	);
+
+	const subCommandGroup = localCommand.options?.find((o) => o.name === commandNames.subCommandGroup) as ApplicationCommandSubGroupData;
+	const subCommand = (subCommandGroup || localCommand)?.options?.find(
+		(o) => o.name === commandNames.subCommand
+	) as ApplicationCommandSubCommandData;
+
+	const commandOptions = [];
+	if (!!subCommand) {
+		if (!!subCommandGroup) {
+			commandOptions.push({
+				name: subCommandGroup.name,
+				type: subCommandGroup.type,
+				options: [
+					{
+						name: subCommand.name,
+						type: subCommand.type,
+						options: formattedOptions,
+					},
+				],
+			});
+		} else {
+			commandOptions.push({
+				name: subCommand.name,
+				type: subCommand.type,
+				options: formattedOptions,
+			});
+		}
+	} else {
+		commandOptions.push(...formattedOptions);
+	}
```

Comment:
Waiting for refactorisation :
Add "subCommandOptions" variable and if subCommandGroup condition is true then wrap subCommandOptions inside another variable

```typescript
if (!!subCommand) {
        // if it's a subcommand we're storing the payload we will be using in case it needs to be reused by subCommandGroup and avoid code repetition
        const subCommandOptions = {
		name: subCommand.name,
		type: subCommand.type,
		options: formattedOptions,
	}
	if (!!subCommandGroup) {
                // if there is a subcommandGroup we encase what we prepared before in a subcommandGroup data format
		subCommandOptions = {
			name: subCommandGroup.name,
			type: subCommandGroup.type,
			options: [subCommandOptions]
		}
	} 
        commandOptions.push(subCommandOptions);
} else {
	commandOptions.push(...formattedOptions);
}
```